notify/impl.go

```diff
@@ -62,6 +66,79 @@ func Build(confs []*config.Receiver, tmpl *template.Template) map[string]Fanout
 
 const contentTypeJSON = "application/json"
 
+// End-users should not be exposed to Go's type system,
+// as this will confuse them and prevent simple things like
+// simple equality checks to fail. Map everything to float64/string.
+type templateData struct {
+	Status                      string
+	Alerts                      []templateAlert
```

Comment:
I would export both types for visibility in the godoc.
This should remain `model.Alerts` in my opinion – of course then labels are no longer strings and such.

I wonder how this would even come up if everything consistently is `model.LabelName` though. Comparisons would work with each other and string literals.
I cannot immediately see rich use-cases that would cause much trouble here.